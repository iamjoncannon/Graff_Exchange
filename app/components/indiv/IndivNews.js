import React from 'react';
import { logoUrl } from '../utils'

export default function IndivNews (props){

  const news  = [
     {title : "blankity blank", date : "1/1/2019", text: "yada yada", image_url: logoUrl }, 
     {title : "blankity blank", date : "1/1/2019", text: "yada yada", image_url: logoUrl }, 
     {title : "blankity blank", date : "1/1/2019", text: "yada yada", image_url: logoUrl }, 
     {title : "blankity blank", date : "1/1/2019", text: "yada yada", image_url: logoUrl }, 
     {title : "blankity blank", date : "1/1/2019", text: "yada yada", image_url: logoUrl }, 
     {title : "blankity blank", date : "1/1/2019", text: "yada yada", image_url: logoUrl }, 
     {title : "blankity blank", date : "1/1/2019", text: "yada yada", image_url: logoUrl }, 
     {title : "blankity blank", date : "1/1/2019", text: "yada yada", image_url: logoUrl }, 
     {title : "blankity blank", date : "1/1/2019", text: "yada yada", image_url: logoUrl }, 
     {title : "blankity blank", date : "1/1/2019", text: "yada yada", image_url: logoUrl }, 
    ];

  return (

    <div className="indiv-container">

      {news.map((newsItem, i) =>{

          return (
              <div className="news-box"> 
                  

                  <div>

                      <span>
                          <a href={newsItem.news_url} target="_blank">
                          {newsItem.title}
                          </a>
                      </span>

                      <span>
                          {newsItem.date}
                      </span>
                      
                      <span>
                          {newsItem.text}
                      </span>
                  
                  </div>

                  <img 
                      src={newsItem.image_url} 
                  />


              </div>
          )
      })

      }
    </div>
  );
};