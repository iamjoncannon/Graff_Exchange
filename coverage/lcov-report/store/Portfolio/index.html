<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for store/Portfolio</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> store/Portfolio
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">47.66% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>61/128</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">16% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>4/25</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">72.73% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>16/22</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">46.67% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>56/120</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file empty" data-value="action_constants_for_Portfolio.js"><a href="action_constants_for_Portfolio.js.html">action_constants_for_Portfolio.js</a></td>
	<td data-value="0" class="pic empty"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	</tr>

<tr>
	<td class="file medium" data-value="actions_for_Portfolio.js"><a href="actions_for_Portfolio.js.html">actions_for_Portfolio.js</a></td>
	<td data-value="77.78" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 77%;"></div><div class="cover-empty" style="width:23%;"></div></div></td>
	<td data-value="77.78" class="pct medium">77.78%</td>
	<td data-value="18" class="abs medium">14/18</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="55.56" class="pct medium">55.56%</td>
	<td data-value="9" class="abs medium">5/9</td>
	<td data-value="77.78" class="pct medium">77.78%</td>
	<td data-value="18" class="abs medium">14/18</td>
	</tr>

<tr>
	<td class="file low" data-value="reducer_for_Portfolio.js"><a href="reducer_for_Portfolio.js.html">reducer_for_Portfolio.js</a></td>
	<td data-value="6.98" class="pic low"><div class="chart"><div class="cover-fill" style="width: 6%;"></div><div class="cover-empty" style="width:94%;"></div></div></td>
	<td data-value="6.98" class="pct low">6.98%</td>
	<td data-value="43" class="abs low">3/43</td>
	<td data-value="10.53" class="pct low">10.53%</td>
	<td data-value="19" class="abs low">2/19</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="7.32" class="pct low">7.32%</td>
	<td data-value="41" class="abs low">3/41</td>
	</tr>

<tr>
	<td class="file medium" data-value="thunks_for_Portfolio.js"><a href="thunks_for_Portfolio.js.html">thunks_for_Portfolio.js</a></td>
	<td data-value="65.67" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 65%;"></div><div class="cover-empty" style="width:35%;"></div></div></td>
	<td data-value="65.67" class="pct medium">65.67%</td>
	<td data-value="67" class="abs medium">44/67</td>
	<td data-value="33.33" class="pct low">33.33%</td>
	<td data-value="6" class="abs low">2/6</td>
	<td data-value="83.33" class="pct high">83.33%</td>
	<td data-value="12" class="abs high">10/12</td>
	<td data-value="63.93" class="pct medium">63.93%</td>
	<td data-value="61" class="abs medium">39/61</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Oct 03 2019 19:08:17 GMT-0400 (Eastern Daylight Time)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
<script src="../../block-navigation.js"></script>
</body>
</html>
