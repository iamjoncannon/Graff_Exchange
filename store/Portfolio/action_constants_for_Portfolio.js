export default {
	HYDRATEPORTFOLIO : 'HYDRATEPORTFOLIO',
	MAKETRADE : 'MAKETRADE',
	GETONEPRICE : 'GETONEPRICE',
	GETOPENINGPRICE : 'GETOPENINGPRICE',

}