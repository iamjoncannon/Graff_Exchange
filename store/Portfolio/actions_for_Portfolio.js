import actions from "./action_constants_for_Portfolio"

const hydratePortfolio = (payload) => {
      	return {
      		type: actions.HYDRATEPORTFOLIO,
      		payload
      	}
      }

const makeTrade = (payload) => {
      	return {
      		type: actions.MAKETRADE,
      		payload
      	}
      }

const getOnePrice = (payload) => {
      	return {
      		type: actions.GETONEPRICE,
      		payload
      	}
      }

const getOpeningPrice = (payload) => {
      	return {
      		type: actions.GETOPENINGPRICE,
      		payload
      	}
      }

export default {
	hydratePortfolio,
	makeTrade,
	getOnePrice,
	getOpeningPrice,
}