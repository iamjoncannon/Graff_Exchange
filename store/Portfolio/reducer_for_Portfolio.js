import actions from "./action_constants_for_PORTFOLIO"

const initialState = {
  selectedPortfolioItem: null,
  portfolio: { },
  transactionHistory: { }
}

export default function Portfolio_reducer (state = initialState, action) {
  
  switch (action.type) {

    case actions.HYDRATEPORTFOLIO: {

      const { portfolio, transactionHistory } = action.payload
      
      let selectedPortfolioItem = !state.selectedPortfolioItem ? portfolio[Object.keys(portfolio)[0]] : selectedPortfolioItem ;

      return { portfolio, transactionHistory, selectedPortfolioItem }
    }
    
    case actions.MAKETRADE: {

      let { symbol, trade } = action.payload

      let updatedPortfolio = {...state.portfolio } 

      updatedPortfolio[symbol]["quantity"] = trade[1]

      return { ...state, portfolio: updatedPortfolio  }
    }
    
    case actions.GETONEPRICE: {

      return { ...state }
    }
    
    case actions.GETOPENINGPRICE: {

      return { ...state }
    }
    
    case actions.HANDLESOCKETMESSAGE: {

      const stock = action.payload

      let updatedPortfolio = {...state.portfolio } 

      updatedPortfolio[stock.symbol]["price"] = stock.price

      return { ...state, portfolio: updatedPortfolio }
    }
    
    case actions.HANDLENEWS: {

      const {symbol, data } = action.payload
      let updatedPortfolio = { ...state.portfolio}

      updatedPortfolio[symbol]["news"] = data.data;
  
      return { ...state, portfolio: updatedPortfolio }
    }
    
    case actions.HANDLEFINANCIALS: {

      const {symbol, financials } = action.payload

      let updatedPortfolio = { ...state.portfolio}

      updatedPortfolio[symbol]["financials"] = financials[0];
  
      return { ...state, portfolio: updatedPortfolio }
    }
    
    case actions.HANDLEHISTORICALPRICE: {

      const {symbol, historical } = action.payload
      
      let updatedPortfolio = { ...state.portfolio}

      updatedPortfolio[symbol]["historical"] = historical;
  
      return { ...state, portfolio: updatedPortfolio }
    }
    
    case actions.HANDLESYMBOLSELECT: {

      const nextSelect = state.portfolio[action.payload]

      return { ...state, selectedPortfolioItem : nextSelect }
    }
    
    default:
      return state
  }
}