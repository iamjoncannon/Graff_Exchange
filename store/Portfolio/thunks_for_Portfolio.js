import actions from "./actions_for_Portfolio"
import { urlPrefix } from '../../secrets'
import axios from 'axios'

function makeHeader(token){
  
  return { headers: {"Authorization": 'Bearer ' + token}}
}

// local storage has no TTL mechanism like Redis,
// so we have to roll our own 

function epoch(){

  // Unix epoch converted hours
  return ( (new Date).getTime() / 1000 ) / (60*60)
}

function setTTL(hours){

  return epoch() + hours
}

function hasTTLExpired(endTime){

  let current = epoch()

  return current > endTime
}

export const hydratePortfolioThunk = (token) => async dispatch => {
 
  let portfolio
  let transactionHistory

  try {

    Promise.all( // allows us to run api calls in parallel rather than serially
      [ 
        ( portfolio = await axios.post( urlPrefix + '/getportfolio', {}, makeHeader(token)) ),
        ( transactionHistory = await axios.post(urlPrefix + '/getallTransactions', {}, makeHeader(token)) )
      ]
    )
  }
  catch(error){

    console.log(error)
  }
  
  portfolio = JSON.parse(portfolio.data)
  
  // prevent nullish errors

  for(let stock in portfolio){
    
    portfolio[stock].data = {}
  }
  
  // update the store with what we have now, so it can
  // start updating the table- this will cause the view
  // to update gradually rather than waiting for all the data
  // to return from the api in the next calls

  dispatch(actions.hydratePortfolio( { portfolio : {...portfolio}, 
                                       transactionHistory: {...JSON.parse(transactionHistory.data)},                 
                                      //  financials : null,
                                      //  historical : null,
                                      //  news : null
                                      } 
                                     ))
  
  // we have the initial portfolio, now we need to add current data
  // from an external API using an endpoint from the Gopher API

  if(portfolio && transactionHistory.data){

    for(let stock in portfolio){
                                        // hoisted below
        portfolio[stock].data = await getOpeningPriceThunk(stock, token)

    }

    dispatch(actions.hydratePortfolio( { portfolio : {...portfolio}, 
                                         transactionHistory: {...JSON.parse(transactionHistory.data)},                 
                                        } ))
  }
  else{

    // registration user story
    // callback({}, {1:{Symbol: "", Quantity: "", Date: ""}})

  }
  
};


// data cached in local storage

export const hydrateSinglePortfolioPage = (token, selectedPortfolioItem) => async dispatch => {

  let { symbol } = selectedPortfolioItem

  // basic pattern of implementing local storage with Unix epoch TTL

  // data for the time series chart

  {

    let inLocalStorage = !!localStorage.getItem(`time-series-${symbol}`)  
    
    let expired = hasTTLExpired(localStorage.getItem(`time-series-${symbol}-TTL`)) 
    
    if( inLocalStorage && !expired ){
      
      dispatch(actions.handleFinancials({ symbol, financials: JSON.parse(localStorage.getItem(`time-series-${symbol}`))}))
    }
    else{

      let data 

      try{

        data = await axios.post( urlPrefix + `/time-series/${symbol}`, {}, makeHeader(token) ) 

      }
      catch(error){

        console.log(error)
      }
        
      if(data){
        
        let { historical } = data.data
        dispatch(actions.handleHistoricalPrice({ symbol, historical }))
        localStorage.setItem(`time-series-${symbol}`, JSON.stringify(historical))
        localStorage.setItem(`time-series-${symbol}-TTL`, setTTL(60*60*24) )
      }        
    }
  }
  
  // data for the news section
  
  {

    let inLocalStorage = !!localStorage.getItem(`news-${symbol}`)  // returns null if key doesn't exist
    
    let expired = hasTTLExpired(localStorage.getItem(`news-${symbol}-TTL`)) 
    
    if( inLocalStorage && !expired ){
      
      dispatch(actions.handleNews({ symbol, data: JSON.parse(localStorage.getItem(`news-${symbol}`))}))
    }
    else{
      
      // calling Gopher API
      let data 
      
      try{
        
        let response = await axios.post( urlPrefix + `/news/${symbol}`, {}, makeHeader(token) ) 

        data = response.data
      }
      catch(error){

        console.log(error)
      }
                
      if(data){
        
          dispatch(actions.handleNews({ symbol, data }))
          localStorage.setItem(`news-${symbol}`, JSON.stringify(data))
          localStorage.setItem(`news-${symbol}-TTL`, setTTL(.30) )
        }
      }
  }


  // data for the quarterly financials section
  {

    let inLocalStorage = !!localStorage.getItem(`financials-${symbol}`)  
    
    let expired = hasTTLExpired(localStorage.getItem(`financials-${symbol}-TTL`)) 
    
    if( inLocalStorage && !expired ){
      
      dispatch(actions.handleFinancials({ symbol, financials: JSON.parse(localStorage.getItem(`financials-${symbol}`))}))
    }
    else{

      let data 

      try{

        data = await axios.post( urlPrefix + `/financials/${symbol}`, {}, makeHeader(token) ) 

      }
      catch(error){

        console.log(error)
      }
        
      if(data){
        
        let { financials } = data.data
        dispatch(actions.handleFinancials({ symbol, financials }))
        localStorage.setItem(`financials-${symbol}`, JSON.stringify(financials))
        localStorage.setItem(`financials-${symbol}-TTL`, setTTL(60*60*24) )
      }        
    }
  }

  

  
  // const timeSeriesData = `https://financialmodelingprep.com/api/v3/historical-price-full/${symbol}?timeseries=365`

  // axios.get(timeSeriesData).then( ({ data }) => {

  //   const { historical } = data

  //   dispatch(actions.handleHistoricalPrice({ symbol, historical }))
  // })

}

export const makeTradeThunk = (symbol, Quantity, Type, Price, token) => async dispatch => {

  /*
    endpoint expects:
     Symbol: '',
     Quantity: '',
     Type: 'Buy',
     Price: ''
  */
 
    Quantity = Number(Quantity)

    let trade 

    try{

      trade = await axios.post( urlPrefix + '/maketransaction', {Symbol: symbol, Quantity, Type, Price}, makeHeader(token)) 
    }
    catch(err){

      console.log(err)
    }

    trade = trade.data

    dispatch(actions.makeTrade({symbol, trade})) 
};

async function getOpeningPriceThunk (symbol, token) {
  
  const url = urlPrefix + '/ohlc/' + symbol

  let returnData  

  try {

    let { data } = await axios.post(url, {}, makeHeader(token))

    returnData = data

  }
  catch(error){
    console.log(error)
  }

  // returned from redis cache

  if(typeof returnData === "string"){

    try{

      returnData = JSON.parse(returnData)
    }
    catch(error){
      console.log("error reading JSON: ", symbol, error)
    }

  }
  
  return returnData
}

export default {
	hydratePortfolioThunk,
	makeTradeThunk,
}
