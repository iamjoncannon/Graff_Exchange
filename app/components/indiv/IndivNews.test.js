import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import IndivNews from './IndivNews'
import configureStore from 'redux-mock-store';
import thunk from "redux-thunk"
const mockStore = configureStore([thunk]);
import { hydrateNewsThunk } from "../../../store/Portfolio/thunks_for_Portfolio"
import { BrowserRouter as Router } from "react-router-dom";
import nocked from "../../../test/nock.setup.js"

const fake_api_data = { 
    data: { 
        hydrate_news : 
                 [{
                date: "Wed, 02 Oct 2019 15:44:44 -0400",
                image_url: "https://cdn.snapi.dev/images/v1/k/0/etf24.jpg",
                news_url: "https://investorplace.com/2019/10/5-top-stock-trades-for-thursday-googl-xom-gm/",
                text: "Alphabet, Exxon Mobil, General Motors, Aphria and the Russell 2000 were our top stock trades to watch going into Thursday.",
                title: "5 Top Stock Trades for Thursday: GOOGL, XOM, GM, APHA & IWM",
                __typename: "News_Story"
                }]
    }
}

const fake_store_without_news_data =  {
 
    Portfolio_state: {
        portfolio: {
            FB: {
                symbol: "FB",
            }
        },
        selectedPortfolioItem: "FB"
    }
}

const fake_store_with_news_data =  {
 
    Portfolio_state: {
        portfolio: {
            FB: {
                symbol: "FB",
                news: fake_api_data.data.hydrate_news
            }
        },
        selectedPortfolioItem: "FB"
    }
}

const fake_store_with_blank_news_data =  {
 
    Portfolio_state: {
        portfolio: {
            FB: {
                symbol: "FB",
                news: []
            }
        },
        selectedPortfolioItem: "FB"
    }
}

const store_setup = (store) => {

    return renderer.create(
        <Provider store={store} >
           <Router >
               <IndivNews location={{pathname: "/"}}/>
           </Router>
       </Provider>
   );
}

describe("IndivNews", ()=>{

    let component 
    let store 

    beforeEach(()=>{
        
        nocked(fake_api_data)
        
        store = mockStore( fake_store_without_news_data );
    
        component = store_setup(store)

    })

    afterEach(()=>{

        store.clearActions();
    })

    it('1) renders successfully with data from redux store', ()=>{

        let tree = component.toJSON();
        
        expect(tree).toMatchSnapshot();
    })

    it("2) if not cell viewport and news not defined, calls hydrateNewsThunk", (done)=>{
       
        store.dispatch( hydrateNewsThunk("FB") )
        .then(()=> {
            
            let dispatched_actions = store.getActions()
            expect(dispatched_actions[0].type).toEqual(dispatched_actions[1].type)    
            done()
        })  
    })

    it("3) it displays loading dots while loading news from API",()=>{

        const target_className = "loading-dots"

        const target = component.root.find(node=> node.props.className === target_className )
    
        expect(target).toBeTruthy()
    })
})

describe("IndivNews", ()=>{

    let component 
    let store 

    beforeEach(()=>{
        
        nocked(fake_api_data)
        
        store = mockStore( fake_store_with_news_data );
    
        component = store_setup(store)

    })

    afterEach(()=>{

        store.clearActions();
    })

    it("4) if news is defined, does not call hydrateNewsThunk", (done)=>{

        store.dispatch( hydrateNewsThunk("FB") )
        .then(()=> {
            
            let dispatched_actions = store.getActions()
            expect(dispatched_actions.length).toEqual(1)    
            done()
        })  
    })

    it("5) renders spans with title, date, and text, and image from image_url", ()=>{

        let news_box = component.root.findAll( node => node.props.className === "news-box")

        expect(news_box.length).toEqual(1)

        let spans = news_box[0].props.children[0].props.children.map(child => child.type) 

        expect(spans.length).toEqual(3)

        expect(news_box[0].props.children[1].type).toEqual("img")
    })
})

describe("IndivNews", ()=>{

    let store 
    let component 

    beforeEach(  () => {

        store = mockStore( fake_store_with_blank_news_data );

        component = store_setup(store)
    });

    afterEach(()=>{

        store.clearActions();
    })
    
    it("6) if no news data from API, renders span saying data not available",  () =>{

        const target = component.root.find(node=> node.props.className === "error_message")
    
        expect(target).toBeTruthy()
    })
})
