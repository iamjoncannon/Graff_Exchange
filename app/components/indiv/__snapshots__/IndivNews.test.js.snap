// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndivNews 1) renders successfully with data from redux store 1`] = `
<div
  className="indiv-container"
>
  <div
    style={
      Object {
        "display": "block",
        "textAlign": "center",
      }
    }
  >
    <div
      className="loading-dots"
    >
      <span
        style={
          Object {
            "color": "black",
            "display": "inline",
            "fontSize": "10vh",
          }
        }
      >
        .
      </span>
      <span
        style={
          Object {
            "color": "black",
            "display": "inline",
            "fontSize": "10vh",
          }
        }
      >
        .
      </span>
      <span
        style={
          Object {
            "color": "black",
            "display": "inline",
            "fontSize": "10vh",
          }
        }
      >
        .
      </span>
    </div>
  </div>
</div>
`;
